src/ctap/pin_protocol_v1.rs

```diff
@@ -328,6 +328,9 @@ impl PinProtocolV1 {
         let token_encryption_key = crypto::aes256::EncryptionKey::new(&shared_secret);
         let pin_decryption_key = crypto::aes256::DecryptionKey::new(&token_encryption_key);
         self.verify_pin_hash_enc(rng, persistent_store, &pin_decryption_key, pin_hash_enc)?;
+        if persistent_store.has_force_pin_change()? {
```

Comment:
Why not moving this test right at the beginning, after the one that checks for pin retries?
It seems from this logic that if invalid PIN is provided, you will return `CTAP2_ERR_PIN_INVALID` from `verify_pin_hash_enc()` 1 line above. If the PIN is valid you'll return `CTAP2_ERR_PIN_INVALID` because of this check. So maybe we can save some time.

Reply:
This is a very good question, and I was wondering why the [specification](https://fidoalliance.org/specs/fido-v2.1-rd-20201208/fido-client-to-authenticator-protocol-v2.1-rd-20201208.html#getPinToken) asks for this behaviour in the order of the protocol. It might be related to allowing to decrement / reset the PIN counter. I added a TODO to investigate, but this is the purpose I can think of.